src/java.base/share/classes/javax/net/ssl/SSLEngine.java

```diff
@@ -336,6 +336,26 @@
  * started, an {@code SSLEngine} can not switch between client and server
  * modes, even when performing renegotiations.
  * <P>
+ * The ApplicationProtocol {@code String} values returned by the methods in
+ * this class are in the network byte representation sent by the peer and
+ * may need to be converted to its Unicode format before use.  For example,
```

Comment:
I think there are two possible directions to use the bytes, concerting application ALPN representation to network byte representation, or concerting network bytes to application representation.  The 1st sentence, I may focus on the the network byte representation description.

"The ApplicationProtocol {@code String} values returned by the methods in this class are in the network byte representation sent by the peer.  If an ALPN value is not encoded in network byte ..., an conversion may be required.  For example, ..."

Reply:
Discussed with Xuelei, the concern was it wasn't clear that you could compare using byte[] or Strings, and possible byte encodings that might be received.  I believe these have been addressed in the current version.